MBBSEmu/HostProcess/HostRoutines/MenuRoutines.cs

```diff
@@ -308,18 +306,29 @@ private void MainMenuInput(SessionBase session, Dictionary<string, MbbsModule> m
                 return;
             }
 
-            var inputValue = Encoding.ASCII.GetString(session.InputBuffer.ToArray()).ToUpper()[0];
+            var inputValue = Encoding.ASCII.GetString(session.InputBuffer.ToArray());
+            
+            //convert to uppercase, trim
+            var inputCommand = inputValue.ToUpper().TrimEnd('\0');
 
+            //Exit if command longer then 2 characters (up to 99 modules) or a negative number
+            if (inputCommand.Length > 2 || inputCommand.Contains('-'))
```

Comment:
This entire section might not be necessary. 

Leave the inputCommand == "X" logic, and then relying on `Int.TryParse` should be sufficient no? Probably have to add an additional check on selectedMenuItem >= 0

Reply:
Added OR statement with >= 0 and removed extra check. Only outstanding item is you can type 00000000 and get the 0 menu item... not sure it matters but that's the only other thing that slips through

Reply:
I'm fine with 00000 still resolving to 0